crates/build/src/docker.rs

```diff
@@ -172,130 +168,40 @@ pub fn docker_build(args: ExecuteArgs) -> Result<BuildResult> {
         })
 }
 
-/// Reads the `BuildResult` produced by the docker execution
-fn read_build_result(
-    host_folder: &Path,
-    build_result_path: &PathBuf,
-) -> Result<BuildResult> {
-    let file = std::fs::File::open(build_result_path)?;
-    let mut build_result: BuildResult =
-        match serde_json::from_reader(BufReader::new(file)) {
-            Ok(result) => result,
-            Err(_) => {
-                // sometimes we cannot remove the file due to privileged access
-                let _ = std::fs::remove_file(build_result_path);
-                anyhow::bail!(
-                    "Error parsing output from docker build. The build probably failed!"
-                )
-            }
-        };
-
-    let new_path = host_folder.join(
-        build_result
-            .target_directory
-            .as_path()
-            .strip_prefix(MOUNT_DIR)?,
-    );
-    build_result.target_directory = new_path;
-
-    let new_path = build_result.dest_wasm.as_ref().map(|p| {
-        host_folder.join(
-            p.as_path()
-                .strip_prefix(MOUNT_DIR)
-                .expect("cannot strip prefix"),
-        )
-    });
-    build_result.dest_wasm = new_path;
-
-    build_result.metadata_result.as_mut().map(|mut m| {
-        m.dest_bundle = host_folder.join(
-            m.dest_bundle
-                .as_path()
-                .strip_prefix(MOUNT_DIR)
-                .expect("cannot strip prefix"),
-        );
-        m.dest_metadata = host_folder.join(
-            m.dest_metadata
-                .as_path()
-                .strip_prefix(MOUNT_DIR)
-                .expect("cannot strip prefix"),
-        );
-        m
-    });
-    Ok(build_result)
-}
-
 /// Overwrites `build_result` and `image` fields in the metadata
 fn update_metadata(
-    build_result: &BuildResult,
+    metadata_artifacts: &MetadataArtifacts,
     verbosity: &Verbosity,
     build_steps: &mut BuildSteps,
-    build_image: &ImageSummary,
+    build_image: &str,
 ) -> Result<()> {
-    if let Some(metadata_artifacts) = &build_result.metadata_result {
-        let mut metadata = ContractMetadata::load(&metadata_artifacts.dest_bundle)?;
-        // find alternative unique identifier of the image, otherwise grab the digest
-        let image_tag = match build_image
-            .repo_tags
-            .iter()
-            .find(|t| !t.ends_with("latest"))
-        {
-            Some(tag) => tag.to_owned(),
-            None => build_image.id.clone(),
-        };
-
-        metadata.image = Some(image_tag);
-
-        crate::metadata::write_metadata(
-            metadata_artifacts,
-            metadata,
-            build_steps,
-            verbosity,
-        )?;
-    }
-    Ok(())
+    let mut metadata = ContractMetadata::load(&metadata_artifacts.dest_bundle)?;
+
+    metadata.image = Some(build_image.to_owned());
```

Comment:
If the `latest` tag is used, you are not actually persisting the version of the image. We need to use only the specific tag